### ISSUE TITLE
kernel problem with wifi dongle Netgear WNA1100 on pre-release ev3dev-jessie-2014-07-12

### ISSUE THREAD

**Issue by BertLindeman** [ev3dev/ev3dev#152]
The Netgear WNA1100 wifi dongle worked fine on [Release 01.00.00 of the ev3dev.img file](https://github.com/ev3dev/ev3dev/releases/tag/b9bdd31)
Now trying on [ev3dev-jessie-2014-07-12](https://github.com/ev3dev/ev3dev/releases/tag/ev3dev-jessie-2014-07-12) 
Problem does not go away with upgrade or dist-upgrade.

It results in very quickly filling various log files, like /var/log/syslog with:
```
Sep 17 10:38:49 ev3dev kernel: usb 1-1: USB disconnect, device number 2
Sep 17 10:38:49 ev3dev kernel: rtw_cmd_thread(wlan0) stop_req:1, break
Sep 17 10:39:30 ev3dev kernel: usb 1-1: new full-speed USB device number 3 using ohci
Sep 17 10:39:30 ev3dev kernel: hub 1-1:1.0: USB hub found
Sep 17 10:39:30 ev3dev kernel: hub 1-1:1.0: 4 ports detected
Sep 17 10:39:31 ev3dev kernel: usb 1-1.2: new full-speed USB device number 4 using ohci
Sep 17 10:39:31 ev3dev kernel: usb 1-1.4: new full-speed USB device number 5 using ohci
Sep 17 10:39:31 ev3dev kernel: usb 1-1.2: ath9k_htc: Firmware htc_9271.fw requested
Sep 17 10:39:32 ev3dev kernel: usbcore: registered new interface driver ath9k_htc
Sep 17 10:39:32 ev3dev kernel: usb 1-1.2: ath9k_htc: Transferred FW: htc_9271.fw, size: 51272
Sep 17 10:39:32 ev3dev kernel: ------------[ cut here ]------------
Sep 17 10:39:32 ev3dev kernel: WARNING: CPU: 0 PID: 13534 at drivers/usb/core/urb.c:450 usb_submit_urb+0x2a4/0x45c()
Sep 17 10:39:32 ev3dev kernel: usb 1-1.2: BOGUS urb xfer, pipe 1 != type 3
Sep 17 10:39:32 ev3dev kernel: Modules linked in: ath9k_htc ath9k_common ath9k_hw ath mac80211 cfg80211 bnep usb_f_eem g_ether usb_f_rndis u_ether libcomposite configfs hci_uart bluetooth rfkill legoev3_uart nfsd i2c_legoev3 nxt_i2c_sensor nxt_i2c_host ev3_uart_host nxt_analog_sensor nxt_light_sensor touch_sensor msensor_class ev3_tacho_motor tacho_motor_class ev3_output_port ev3_input_port suart_emu 8192cu(O) snd_legoev3 snd_pcm snd_timer snd soundcore legoev3_ports legoev3_battery legoev3_bluetooth legoev3_analog
Sep 17 10:39:32 ev3dev kernel: CPU: 0 PID: 13534 Comm: kworker/0:2 Tainted: G           O  3.16.1-2-ev3dev #2
Sep 17 10:39:32 ev3dev kernel: Workqueue: events request_firmware_work_func
Sep 17 10:39:32 ev3dev kernel: Backtrace: 
Sep 17 10:39:32 ev3dev kernel: [<c000c088>] (dump_backtrace) from [<c000c228>] (show_stack+0x18/0x1c)
Sep 17 10:39:32 ev3dev kernel: r6:c0497951 r5:00000009 r4:c1f2dcb8 r3:60000013
Sep 17 10:39:32 ev3dev kernel: [<c000c210>] (show_stack) from [<c03b27f0>] (dump_stack+0x20/0x28)
Sep 17 10:39:32 ev3dev kernel: [<c03b27d0>] (dump_stack) from [<c0019fe8>] (warn_slowpath_common+0x6c/0x8c)
Sep 17 10:39:32 ev3dev kernel: [<c0019f7c>] (warn_slowpath_common) from [<c001a0ac>] (warn_slowpath_fmt+0x38/0x40)
Sep 17 10:39:32 ev3dev kernel: r8:00000001 r7:00000002 r6:00000040 r5:c199d000 r4:c3af8a80
Sep 17 10:39:32 ev3dev kernel: [<c001a078>] (warn_slowpath_fmt) from [<c026c6ec>] (usb_submit_urb+0x2a4/0x45c)
Sep 17 10:39:32 ev3dev kernel: r3:c047988c r2:c0497987
Sep 17 10:39:32 ev3dev kernel: [<c026c448>] (usb_submit_urb) from [<bf365fec>] (hif_usb_send+0x378/0x3e0 [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r10:c2e34008 r9:00009030 r8:c3961274 r7:c360b1c0 r6:c3af8a80 r5:c2e10600
Sep 17 10:39:32 ev3dev kernel: r4:c2f1f180
Sep 17 10:39:32 ev3dev kernel: [<bf365c74>] (hif_usb_send [ath9k_htc]) from [<bf364068>] (htc_issue_send.clone.2+0x68/0x6c [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r8:c3961274 r7:c3961000 r6:0000000a r5:c2e10600 r4:c3961000 r3:bf36e340
Sep 17 10:39:32 ev3dev kernel: [<bf364000>] (htc_issue_send.clone.2 [ath9k_htc]) from [<bf36438c>] (htc_connect_service+0x194/0x1f0 [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r7:c396124c r6:c2e10600 r5:c1f2dd94 r4:c3961000
Sep 17 10:39:32 ev3dev kernel: [<bf3641f8>] (htc_connect_service [ath9k_htc]) from [<bf366dc4>] (ath9k_wmi_connect+0x58/0x74 [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r10:00000000 r8:00000000 r7:00000000 r6:c0940cd8 r5:c3961000 r4:c2e34000
Sep 17 10:39:32 ev3dev kernel: [<bf366d6c>] (ath9k_wmi_connect [ath9k_htc]) from [<bf36c33c>] (ath9k_init_htc_services.clone.5+0x28/0x288 [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r6:c2dbf420 r5:c3961000 r4:c0940cc0
Sep 17 10:39:32 ev3dev kernel: [<bf36c314>] (ath9k_init_htc_services.clone.5 [ath9k_htc]) from [<bf36c844>] (ath9k_htc_probe_device+0x88/0x794 [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r8:c0940340 r7:00000000 r6:c2dbf420 r5:c3961000 r4:c0940cc0
Sep 17 10:39:32 ev3dev kernel: [<bf36c7bc>] (ath9k_htc_probe_device [ath9k_htc]) from [<bf3648f4>] (ath9k_htc_hw_init+0x1c/0x3c [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: r10:00000000 r9:00000000 r8:c051975c r7:c12b7a40 r6:00000000 r5:c2068c20
Sep 17 10:39:32 ev3dev kernel: r4:c2f1f180
Sep 17 10:39:32 ev3dev kernel: [<bf3648d8>] (ath9k_htc_hw_init [ath9k_htc]) from [<bf366770>] (ath9k_hif_usb_firmware_cb+0x14c/0x170 [ath9k_htc])
Sep 17 10:39:32 ev3dev kernel: [<bf366624>] (ath9k_hif_usb_firmware_cb [ath9k_htc]) from [<c023a1a8>] (request_firmware_work_func+0x38/0x58)
Sep 17 10:39:32 ev3dev kernel: r6:00000000 r5:c3f71700 r4:c12b7a40
Sep 17 10:39:32 ev3dev kernel: [<c023a170>] (request_firmware_work_func) from [<c002fc80>] (process_one_work+0x234/0x380)
Sep 17 10:39:32 ev3dev kernel: r4:c2caa660
Sep 17 10:39:32 ev3dev kernel: [<c002fa4c>] (process_one_work) from [<c0031120>] (worker_thread+0x420/0x560)
Sep 17 10:39:32 ev3dev kernel: r10:00000008 r9:00000000 r8:00000000 r7:c051975c r6:c2caa678 r5:c051975c
Sep 17 10:39:32 ev3dev kernel: r4:c2caa660
Sep 17 10:39:32 ev3dev kernel: [<c0030d00>] (worker_thread) from [<c00362fc>] (kthread+0xcc/0xe0)
Sep 17 10:39:32 ev3dev kernel: r10:00000000 r9:00000000 r8:00000000 r7:c0030d00 r6:c2caa660 r5:00000000
Sep 17 10:39:32 ev3dev kernel: r4:c3155200 r3:00000000
Sep 17 10:39:32 ev3dev kernel: [<c0036230>] (kthread) from [<c00095f0>] (ret_from_fork+0x14/0x24)
Sep 17 10:39:32 ev3dev kernel: r7:00000000 r6:00000000 r5:c0036230 r4:c3155200
Sep 17 10:39:32 ev3dev kernel: ---[ end trace 067a9e6629bd4721 ]---
Sep 17 10:39:32 ev3dev kernel: ------------[ cut here ]------------
```

I have an Edimax dongle doing fine on jessie, so for me no real problem at the moment.

**Comment by dlech** [comment]
Thanks to the advice of @cunctator on ev3dev/ev3dev-kernel#5, the ev3dev kernel has been updated to the latest upstream 3.16 series kernel which finally fixes this issue.

The original issue of log flooding was fixed by the revert of 33977be and the additional issue of the WNA1100 still not working correctly after this was fixed in one of the 1000+ upstream commits between 3.16.1 and 3.16.7.

Fix is released in kernel v3.16.7-ckt3-ev3dev1, so upgrade your kernels.

**Comment by BertLindeman** [comment]
Confirmed working OK on 3.16.7-ckt3-ev3dev1
